sdk/python/tests/unit/test_offline_server.py

```diff
@@ -0,0 +1,235 @@
+import os
+import tempfile
+from datetime import datetime, timedelta
+
+import assertpy
+import pandas as pd
+import pyarrow as pa
+import pyarrow.flight as flight
+import pytest
+
+from feast import FeatureStore
+from feast.infra.offline_stores.remote import (
+    RemoteOfflineStore,
+    RemoteOfflineStoreConfig,
+)
+from feast.offline_server import OfflineServer
+from feast.repo_config import RepoConfig
+from tests.utils.cli_repo_creator import CliRunner
+
+PROJECT_NAME = "test_remote_offline"
+
+
+@pytest.fixture
+def empty_offline_server(environment):
+    store = environment.feature_store
+
+    location = "grpc+tcp://localhost:0"
+    return OfflineServer(store=store, location=location)
+
+
+@pytest.fixture
+def arrow_client(empty_offline_server):
+    return flight.FlightClient(f"grpc://localhost:{empty_offline_server.port}")
+
+
+def test_offline_server_is_alive(environment, empty_offline_server, arrow_client):
+    server = empty_offline_server
+    client = arrow_client
+
+    assertpy.assert_that(server).is_not_none
+    assertpy.assert_that(server.port).is_not_equal_to(0)
+
+    actions = list(client.list_actions())
+    flights = list(client.list_flights())
+
+    assertpy.assert_that(actions).is_empty()
+    assertpy.assert_that(flights).is_empty()
+
+
+def default_store(temp_dir):
+    runner = CliRunner()
+    result = runner.run(["init", PROJECT_NAME], cwd=temp_dir)
+    repo_path = os.path.join(temp_dir, PROJECT_NAME, "feature_repo")
+    assert result.returncode == 0
+
+    result = runner.run(["--chdir", repo_path, "apply"], cwd=temp_dir)
+    assert result.returncode == 0
+
+    fs = FeatureStore(repo_path=repo_path)
+    return fs
+
+
+def remote_feature_store(offline_server):
+    offline_config = RemoteOfflineStoreConfig(host="0.0.0.0", port=offline_server.port)
```

Comment:
can we mention type here as a `remote`, just for clarity I know by default it's assigned remote WDYT